### ISSUE TITLE
SAPUI5 User Interface

### ISSUE THREAD

**Issue by voxron** [SAPDocuments/Tutorials#4923]
Tutorial URL: https://developers.sap.com/tutorials/xsa-sapui5.html

Please specify the step you are referring to

Please check all exercies, because in code we have reference to odataTest, but it never has been created. Also we can look odataTest.xml. If we will following steps, so that not done successful.
Also, how can I get to know provider-id : com.sap.ui5.dist.sapui5-dist-xsa.XSAC_UI5_FESV3:sapui5_fesv3???
If I have SAP HANA, and WebIde, how can I get correct link?

**Comment by jung-thomas** [comment]
I'm afraid that many of these HANA Express Tutorials are massively outdated. And there appears to be duplicate tutorials that cover the same topics.  I've been working to clean them up and some like this one might well be removed.  I'd suggest you instead go through this mission: https://developers.sap.com/mission.xsa-get-started.html

It was all fully updated and tested on HANA 2.0 SPS 04 just last week.  Specifically within there we have this tutorial:
https://developers.sap.com/tutorials/xsa-html5-expand-module.html
It shows the correct way to reference the UI5 service broker unlike the tutorial you were going through.

**Comment by voxron** [comment]
Hi,

Thank you for the links! But I've done it. My goal now to understand how to create UI web-page, and create new DB element with using this page. That's why I try to do exercies by the first link, because, it expalin that.

**Comment by jung-thomas** [comment]
I have updated xsa-sapui5.html now as well.  I've moved the parts about the UI5 service broker over to the xsa-html5-expand-module tutorial.  Everything else has been updated for HANA 2.0 SPS04 and retested.  The parts that reference odataTest are correct.  There is nothing created with the odataTest name.  However look at the code in index.html:

```
		data-sap-ui-resourceroots='{
        		"common": "../common",
				"root": "./",
				"opensap.odataTest": "./"}'	
```
That line basically creates an alias named opensap.odataTest that is mapped to the root folder of this project. This way the views, controller, etc can all be written for this alias and not the actual folder names. This allows for easier changes and some reuse without having to rename these references.
